code/modules/mob/living/silicon/laws.dm

```diff
@@ -98,3 +106,31 @@
 	laws_sanity_check()
 	laws.clear_law_sixsixsix(force)
 	post_lawchange(announce)
+
+/mob/living/silicon/proc/get_lawset_name()
+	if(!laws)
+		return
+
+	var/list/law_list = laws.get_law_list(!is_malf(), FALSE, FALSE)
+
+	if(!law_list.len)
+		return "None"
+
+	var/datum/ai_laws/lawset_datum = null
+
+	for(var/lawset in subtypesof(/datum/ai_laws))
+		lawset_datum = new lawset
+
+		if(compare_list(law_list, lawset_datum.get_law_list(TRUE, FALSE, FALSE)))
+			return lawset_datum.name
```

Comment:
This is pretty inefficient. Cache it on a global associative list instead.
Come around the coder channels if you don't know how to do that.